### PR TITLE
Qute - fix incorrect behavior of Evaluator in loop sections

### LINKED ISSUE #10572: Problem with Qute context using for and each blocks
**Describe the bug**
Using `for` and `each` blocks, when accessing an existing key, I think it shouldn't use the `data` context no?

Reproducer code:
```Java
    @Test
    void quteTest() throws IOException {
        final HashMap<String, Object> dep1 = new HashMap<>();
        dep1.put("version", "1.0");

        final HashMap<String, Object> data = new HashMap<>();
        data.put("dependencies", Arrays.asList(dep1, new HashMap<>()));
        data.put("version", "hellllllo");
        final String result = Engine.builder().addDefaults().build().parse("{#for dep in dependencies}\n{#if dep.version}<version>{dep.version}</version>\n{/if}\n{/for}").render(data);
        Assertions.assertThat(result).doesNotContain("hellllllo");
    }
```

Maybe this is an expected behavior with `it` using a `each` block, but with `for` assigned to `dep` in the example it doesn't make sense at all.

### PR DESCRIPTION
**Pull request by mkouba**
- only try the parent context/outer scope for the first part of an
expression
- resolves #10572

### CODE REVIEW

File: independent-projects/qute/core/src/test/java/io/quarkus/qute/LoopSectionTest.java [hunk 1]
```diff
@@ -147,4 +148,17 @@ public void testNoniterable() {
...
+        dep1.put("version", "1.0");
+        final HashMap<String, Object> data = new HashMap<>();
+        data.put("dependencies", Arrays.asList(dep1, new HashMap<>()));
+        data.put("version", "hellllllo");
+        final String result = Engine.builder().addDefaults().build()
+                .parse("{#for dep in dependencies}{#if dep.version}<version>{dep.version}</version>{/if}{/for}")
+                .render(data);
+        assertFalse(result.contains("hellllllo"), result);
+    }
+
```
Comment (ia3andy) [inline independent-projects/qute/core/src/test/java/io/quarkus/qute/LoopSectionTest.java:163]:
```suggestion

    @Test
    void testScopeEach() {
        final HashMap<String, Object> dep1 = new HashMap<>();
        dep1.put("version", "1.0");
        final HashMap<String, Object> data = new HashMap<>();
        data.put("dependencies", Arrays.asList(dep1, new HashMap<>()));
        data.put("version", "hellllllo");
        final String result = Engine.builder().addDefaults().build()
                .parse("{#each dependencies}{#if it.version}<version>{it.version}</version>{/if}{/each}")
                .render(data);
        assertFalse(result.contains("hellllllo"), result);
    }
```

**Review by ia3andy — APPROVED** [review]
LGTM

I would add the suggested test also.

File: independent-projects/qute/core/src/test/java/io/quarkus/qute/LoopSectionTest.java (hunk 1 above)
Comment (ia3andy) [reply independent-projects/qute/core/src/test/java/io/quarkus/qute/LoopSectionTest.java:163]: Oh and maybe we should also test that `{version}` works in another test?

**Comment by mkouba** [comment]
> Oh and maybe we should also test that {version} works in another test?

@ia3andy Something like we have now?

**Review by ia3andy — APPROVED** [review]
Awesome!

**Pull request merged**